src/genie_python/block_names.py

```diff
@@ -1,13 +1,20 @@
 import zlib
 from keyword import iskeyword
 from threading import RLock, Timer
+from typing import TypeVar
+
+import numpy as np
+import numpy.typing as npt
 
 from .channel_access_exceptions import UnableToConnectToPVException
 from .genie_blockserver import BLOCK_SERVER_PREFIX, PV_BLOCK_NAMES
 from .genie_cachannel_wrapper import CaChannelWrapper
 from .utilities import dehex_decompress_and_dejson
 
 DELAY_BEFORE_RETRYING_BLOCK_NAMES_PV_ON_FAIL = 30.0
+E = TypeVar("E", bound=np.generic, covariant=True)
+PVBaseValue = bool | int | float | str
+PVValue = PVBaseValue | list[PVBaseValue] | npt.NDArray[E] | None
```

Comment:
PVValue is already defined in genie, and should be imported from there if possible e.g.
```
 if TYPE_CHECKING:
    from genie_python.genie import PVValue
```
This is how its imported in `genie_pv_connection_protocol.py` for example